### PR TITLE
gossip: support signed map roots from Trillian

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

**Comment by daviddrysdale** [comment]
Might be easiest reviewed commit-by-commit.

File: gossip/client/hubclient_test.go [hunk 1]
```diff
@@ -597,6 +597,11 @@ func TestGetSourceKeys(t *testing.T) {
 			body:     fmt.Sprintf(`{"entries":[{"id":"https://a-log.com","pub_key":"%s","kind":"TRILLIANSLR"}]}`, b64(pubKeyDER)),
 			wantKind: "TRILLIANSLR",
 		},
+		{
+			desc:     "Valid Trillian SMR Source",
+			body:     fmt.Sprintf(`{"entries":[{"id":"https://a-log.com","pub_key":"%s","kind":"TRILLIANSMR"}]}`, b64(pubKeyDER)),
```
Comment (gdbelvin) [inline gossip/client/hubclient_test.go]: a-map.com ?

File: gossip/hub/handlers.go [hunk 2]
```diff
@@ -119,6 +120,19 @@ func trillianSLRChecker(data, sig []byte) error {
 	return nil
 }
 
+func trillianSMRChecker(data, sig []byte) error {
```
Comment (gdbelvin) [inline gossip/hub/handlers.go]: This function looks like [VerifySignedMapRoot](https://github.com/google/trillian/blob/master/crypto/verifier.go#L48). Would that be better?

Comment (gdbelvin) [reply gossip/hub/handlers.go]: `sig` is unused. Is that intentional?

File: gossip/client/hubclient_test.go (hunk 1 above)
Comment (daviddrysdale) [reply gossip/client/hubclient_test.go]: The tests all share the same source URL, so changed it to `source.example.com`

File: gossip/hub/handlers.go (hunk 2 above)
Comment (daviddrysdale) [reply gossip/hub/handlers.go]: The signature gets checked (in `addSignedBlob`) before any of the `submissionCheck` instances get called, so not checking `sig` is intentional. 

I included `sig` in the prototype for the structure checkers to allow for the possibility of a structure check that involving the signature, but given there are three now and none of them use it, I suspect YAGNI.  I'll drop it from the `submissionCheck` signature.

**Comment by gdbelvin** [comment]
Perhaps use the `_` operator then?

I think the `golangci-lint` `unparam` linter will catch this sort of thing.

**Pull request merged**